src/components/CompareResults/ResultsTitle.tsx

```diff
@@ -0,0 +1,87 @@
+import { useState } from 'react';
+
+import EditIcon from '@mui/icons-material/EditOutlined';
+import { Input } from '@mui/material';
+import { style } from 'typestyle';
+
+import useRawSearchParams from '../../hooks/useRawSearchParams';
+import { FontsRaw, Colors } from '../../styles';
+
+export const ResultsTitle = () => {
+  const [searchParams, updateSearchParams] = useRawSearchParams();
+  const [resultsTitle, setResultsTitle] = useState(
+    searchParams.get('title') || 'Results',
+  );
+  const [isEditing, setIsEditing] = useState(false);
+
+  const styles = {
+    title: style({
+      ...FontsRaw.HeadingXS,
+      fontWeight: 700,
+      letterSpacing: '-0.01em',
+      margin: 0,
+    }),
+    editIcon: style({
+      marginLeft: '10px',
+      cursor: 'pointer',
+      color: Colors.LinkText,
+    }),
+    hide: style({
+      display: 'none',
+    }),
+  };
+
+  const handleResultsChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newTitle = event.target.value.trim();
+    setResultsTitle(newTitle);
+  };
+
+  const handleEdit = () => setIsEditing(true);
+
+  const handleEditComplete = (
+    event: React.FocusEvent | React.KeyboardEvent,
+  ) => {
+    if (
+      (event as React.KeyboardEvent).key === 'Enter' ||
+      event.type === 'blur'
+    ) {
+      const safeTitle = resultsTitle || 'Results';
+      setResultsTitle(safeTitle);
+
+      const newSearchParams = new URLSearchParams(searchParams);
+      newSearchParams.set('title', safeTitle);
+      updateSearchParams(newSearchParams, {
+        method: 'push',
+      });
+
+      setIsEditing(false);
+    }
+  };
+
+  return (
+    <>
+      <div className={isEditing ? styles.hide : styles.title}>
```

Comment:
instead of using a `styles.hide`, it's easier and more readable in this case to do a ternary:
```js
{isEditing ? <Input ... > : <div>...</div>}
```

This way it's super clear that we have one or the other.